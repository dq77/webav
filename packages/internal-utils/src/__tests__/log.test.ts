import { vi, expect, test, beforeEach } from 'vitest';
import { Log } from '../log';

const errorSpy = vi.spyOn(console, 'error');
const warnSpy = vi.spyOn(console, 'warn');
const infoSpy = vi.spyOn(console, 'info');
const debugSpy = vi.spyOn(console, 'debug');

beforeEach(() => {
  Log.setLogLevel(Log.info);
  errorSpy.mockReset();
  warnSpy.mockReset();
  infoSpy.mockReset();
  debugSpy.mockReset();
});

test('dump log', async () => {
  const spys = (
    ['getHours', 'getMinutes', 'getSeconds', 'getMilliseconds'] as const
  ).map((methodName) => {
    const spy = vi.spyOn(Date.prototype, methodName);
    spy.mockReturnValue(0);
    return spy;
  });

  const historyStr = await Log.dump();
  Log.setLogLevel(Log.info);
  Log.info('log info');
  Log.warn('log warn');
  Log.error('log error');

  expect((await Log.dump()).replace(historyStr, '')).toMatchSnapshot();
  for (const s of spys) s.mockRestore();
});

test('log threshold', () => {
  Log.setLogLevel(Log.debug);
  Log.debug('log test');
  expect(debugSpy).toHaveBeenCalledWith('log test');
  debugSpy.mockReset();

  Log.setLogLevel(Log.error);
  Log.debug('log test');
  Log.info('log test');
  Log.warn('log test');
  Log.error('log test');
  expect(errorSpy).toHaveBeenCalledWith('log test');
  expect(warnSpy).not.toHaveBeenCalled();
  expect(infoSpy).not.toHaveBeenCalled();
  expect(debugSpy).not.toHaveBeenCalled();
});
