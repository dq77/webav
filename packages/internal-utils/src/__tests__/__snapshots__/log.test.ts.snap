// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`dump log 1`] = `
"[info][0:0:0.0]  log info
[warn][0:0:0.0]  log warn
[error][0:0:0.0]  log error
"
`;
